recipes/gnuradio/meta.yaml

```
@@ -0,0 +1,275 @@
+    requirements:
+      build:
+        - {{ compiler('c') }}
+        - {{ compiler('cxx') }}
+        - cmake
+        - ninja
+        - pkg-config  # [not win]
+      host:
+        - alsa-lib  # [linux]
+        - boost-cpp
+        - fftw
+        - gsl
+        - libblas
+        - libcblas
+        - libusb
+        - log4cpp
+        - mpir
+        - portaudio  # [linux]
+        - python
+        - swig
+        - volk
+      run:
+        - boost-cpp
+        - click
+        - click-plugins
+        - fftw
+        - {{ pin_compatible('log4cpp') }}
+        - mako
+        - numpy
+        - {{ pin_compatible('portaudio') }}  # [linux]
+        - python
+        - six
+    test:
+      commands:
+        - gnuradio-config-info -v --prefix --sysconfdir --prefsdir --userprefsdir --prefs --builddate --enabled-components --cc --cxx --cflags
+        # run a few example scripts to show that at least some things work
+        - python "$CONDA_PREFIX/share/gnuradio/examples/volk_benchmark/volk_math.py" -L test -N 1e6 -I 3 --all  # [not win]
+        - python "%CONDA_PREFIX%/Library/share/gnuradio/examples/volk_benchmark/volk_math.py" -L test -N 1e6 -I 3 --all  # [win]
+      imports:
+        - gnuradio.analog
+        - gnuradio.audio
+        - gnuradio.blocks
+        - gnuradio.channels
+        - gnuradio.digital
+        - gnuradio.dtv
+        - gnuradio.fec
+        - gnuradio.fft
+        - gnuradio.filter
+        - gnuradio.gr
+        - gnuradio.trellis
+        - gnuradio.vocoder
+        - gnuradio.wavelet
+        - pmt
+    about:
+      summary: 'GNU Radio core functionality and modules'
+  - name: gnuradio-qtgui
+    script: install_qtgui.sh  # [not win]
+    script: install_qtgui.bat  # [win]
+    requirements:
+      build:
+        - {{ compiler('c') }}
+        - {{ compiler('cxx') }}
+        - cmake
+        - ninja
+        - pkg-config  # [not win]
+      host:
+        - gnuradio-core
+        - boost-cpp
+        - log4cpp
+        - pyqt
+        - python
+        - qt
+        - qwt
+        - volk
+      run:
+        - {{ pin_subpackage('gnuradio-core', min_pin='x.x.x', max_pin='x.x.x') }}
+        - boost-cpp
+        - {{ pin_compatible('log4cpp') }}
+        - pyqt
+        - python
+        - {{ pin_compatible('qwt', max_pin='x.x') }}
+    test:
+      imports:
+        - gnuradio.qtgui
+    about:
+      summary: 'GNU Radio QT module providing graphical components'
+  - name: gnuradio-uhd
+    script: install_uhd.sh  # [not win]
+    script: install_uhd.bat  # [win]
+    requirements:
+      build:
+        - {{ compiler('c') }}
+        - {{ compiler('cxx') }}
+        - cmake
+        - ninja
+        - pkg-config  # [not win]
+      host:
+        - gnuradio-core
+        - boost-cpp
+        - log4cpp
+        - python
+        - uhd
+      run:
+        - {{ pin_subpackage('gnuradio-core', min_pin='x.x.x', max_pin='x.x.x') }}
```

Perhaps this is what you want in case changes are made by build number:
```suggestion
        - {{ pin_subpackage('gnuradio-core', exact=True) }}
```
(also applies to the other places)

---

The subpackages should be compatible between ``x.x.x`` versions, so my thought was to represent that accurately in the pin and not over-constrain it. Third party gnuradio modules exist, and I could imagine them one day being packaged for conda-forge using an ``x.x.x`` pin to slightly limit rebuilds. This would make the internal modules work similarly, although being outputs from the same package it's not saving any work. I don't feel strongly either way, so maybe it is better to just keep them in sync by build number as well. Any thoughts after hearing that rationale?

---

That's a valid reason and I'm happy for you to skip this suggestion if you'd prefer.

(deleted my initial reply) I've just released `gnuradio` is pinned exactly so most of the issues are avoided. It might be able to get into a broken state by installing different builds of the components directly but most people probably won't do that so it will be fine.

---

Hmm, with further consideration I don't think having a slightly looser pin would provide any real benefits here and might only end up causing headaches. Better to do the simple thing.